### PR TITLE
[torchax] Fix functional.max_pool

### LINKED ISSUE #8086: [hard] `nn.functional.max_pool2d` and `nn.functional.max_pool3d`
## Fix the Op info test for `nn.functional.max_pool2d` and `nn.functional.max_pool3d`

1. Find the line of [test_ops.py](test/test_ops.py) and remove
   `nn.functional.max_pool2d` and `nn.functional.max_pool3d` from `skip_list`
2. Run op_info test with `pytest test/test_ops.py`
3. Fix the failure.

Please refer to 
[this guide](https://github.com/pytorch/xla/blob/master/experimental/torch_xla2/docs/fixing_op_info_test.md) as guide to fix:

Also refer to these PRs:
* https://github.com/pytorch/xla/pull/7343
* https://github.com/pytorch/xla/pull/7312
* https://github.com/pytorch/xla/pull/7325

### PR DESCRIPTION
**Pull request by dvhg**
Fix #8086.

Also possibly related to #8241.

To summarize the changes to max pool:
1. Dilation now correctly accounted for, including in padding calculation following the formula for output shape from [here](https://pytorch.org/docs/stable/generated/torch.nn.MaxPool2d.html)
2. Default stride is equal to kernel size
3. Indices are computed per-batch instead of across the entire input
4. Ties break towards higher indices (i.e. pick the higher index if there is more than 1 max value in a window). This is arbitrary but matches Torch's behavior on CPU.

### CODE REVIEW

**Review by ManfeiBai — APPROVED** [review]
Thanks, LGTM

**Review by qihqi — APPROVED** [review]

**Review by qihqi — APPROVED** [review]
